RX_FSK/data/livemap.js

```diff
@@ -0,0 +1,396 @@
+    var stat;
+    if (data.id) {
+
+      if ((data.lat != '0.000000' && data.lon != '0.000000') && (JSON.stringify(data) !=  JSON.stringify(last_data)) ) {
+        var location = [data.lat,data.lon,data.alt];
+        if (!marker) {
+          map.setView(location, 14);
+          marker = L.marker(location).addTo(map)
+          .bindPopup(poptxt('position',data),{closeOnClick:false, autoPan:false}).openPopup();
+          get_predict(data);
+        } else {
+          marker.slideTo(location, {
+              duration: 500,
+              keepAtCenter: (follow=='marker')?true:false
+          })
+          .setPopupContent(poptxt('position',data));
+          if (last_data.id != data.id) {
+            storage_remove();
+            dots = [];
+            get_predict(data);
+          }
+        }
+        dots.push(location);
+        line.setLatLngs(dots);
+        storage_write(data);
+        $('#status').html('🟢');
+        stat = 1;
+      } else {
+        $('#status').html('🟡');
+        stat = 0;
+      }
+      headtxt(data,stat);
+      last_data = data;
+    } else {
+      $('#status').html('🟡');
+      headtxt(data,0);
+    }
+  };
+
+
+  marker_gps = false;
+  icon_gps = L.divIcon({className: 'leaflet-gps'});
+  circ_gps = false;
+
+  gps = function(e) {
+    gps_location = [e.lat/1000000,e.lon/1000000];
+    gps_accuracy = e.hdop*2;
+
+    if (last_data && last_data.lat != '0.000000') {
+      if ($('.leaflet-footer').css('display') == 'none') { $('.leaflet-footer').show(); }
+
+      var distance = Math.round(map.distance(gps_location,[last_data.lat, last_data.lon]));
+      distance = (distance > 1000)?(distance / 1000) + 'k':distance;
+      $('.leaflet-footer .gps_dist').html(distance);
+
+      $('.leaflet-footer .gps_dir').html( bearing(gps_location,[last_data.lat, last_data.lon]) );
+    }
+
+    if (!marker_gps) {
+      map.addControl(new L.Control.Button([{ position: 'topleft', text: '🛰️', href: 'javascript:show(marker_gps,\'gps\');' }]));
+
+      marker_gps = L.marker(gps_location,{icon:icon_gps}).addTo(map)
+      .bindPopup(poptxt('gps',e),{closeOnClick:false, autoPan:false});
+      circ_gps = L.circle(gps_location, gps_accuracy).addTo(map);
+    } else {
+      marker_gps.slideTo(gps_location, {
+          duration: 500,
+          keepAtCenter: (follow=='gps')?true:false
+      })
+      .setPopupContent(poptxt('gps',e));
+      circ_gps.slideTo(gps_location, { duration: 500 });
+      circ_gps.setRadius(gps_accuracy);
+    }
+  };
+
+  get_data = function() {
+      $('#status').html('🔴');
+      $.get('live.json', function( data ) {
+        if (typeof data != "object") { data = $.parseJSON(data); }
+        if (data.sonde) {
+          draw(data.sonde);
+        } else {
+          setTimeout(function() {$('#status').html('🟡');},100);
+        }
+        if (data.gps) {
+          gps(data.gps);
+        }
+      });
+  };
+
+  predictor = false;
+  get_predict = function(data) {
```

Comment:
Hey, do you know if the latitude is in the range (0 - 180) or (-90 to +90)?

I'm just checking in reference to #142

Reply:
Source of this is:

`SondeInfo *s = &sonde.sondeList[sonde.currentSonde];`

I think @dl9rdz knows better than me.

Reply:
I guess this is a longitude issue, not latitude.

My software uses -180 to 180 and so does leaflet, but the tawhiri API uses 0 - 360. I noted that problem with my Android App earlier this year when I was on Iceland. I guess we will need a similar fix for the livemap
https://github.com/dl9rdz/rdzwx-go/commit/0fe5f00751671c8bdd4fa1f9d534b7c24f3378d4

Reply:
Fixed, see pull request: https://github.com/dl9rdz/rdz_ttgo_sonde/pull/148